co-simulation/lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/transform/Proj4Projection.java

```diff
@@ -0,0 +1,195 @@
+
+    }
+
+    @Override
+    public Vector3d geographicToVector(GeoPoint geographic, Vector3d result){
+        getGeoCalculator().distanceBetween(geoOrigin, geographic, result);
+        return result;
+    }
+
+    @Override
+    public MutableGeoPoint vectorToGeographic(Vector3d vector3d, MutableGeoPoint result) {
+        getGeoCalculator().pointFromDirection(geoOrigin, vector3d, result);
+        return result;
+    }
+
+    @Override
+    public MutableCartesianPoint geographicToCartesian(GeoPoint geographic, MutableCartesianPoint result) {
+
+        ProjCoordinate sourceCoord = new ProjCoordinate(geographic.getLongitude(),geographic.getLatitude());
+        ProjCoordinate targetCoord = new ProjCoordinate();
+        CoordinateTransformFactory ctFactory = new CoordinateTransformFactory();
+        CoordinateTransform transform = ctFactory.createTransform(wgs84CRS, cartesianCRS);
+
+        transform.transform(sourceCoord, targetCoord);
+        result.set(targetCoord.x, targetCoord.y, 0.0);
+
+        return result;
+    }
+
+    @Override
+    public MutableGeoPoint cartesianToGeographic(CartesianPoint cartesian, MutableGeoPoint result) {
+
+        ProjCoordinate sourceCoord = new ProjCoordinate(cartesian.getX() - this.x_offset, cartesian.getY() - this.y_offset);
+        ProjCoordinate targetCoord = new ProjCoordinate();
+        CoordinateTransformFactory ctFactory = new CoordinateTransformFactory();
+        CoordinateTransform transform = ctFactory.createTransform(cartesianCRS, wgs84CRS);
+
+        transform.transform(sourceCoord, targetCoord);
+        //Transform returned as lat,lon = (targetCoord.y, targetCoord.x)
+        result.set(targetCoord.y, targetCoord.x, 0.0);
+
+        return result;
+    }
+
+    @Override
+    public Vector3d utmToVector(UtmPoint utm, Vector3d result) {
+        return geographicToVector(utmToGeographic(utm));
+    }
+
+    @Override
+    public MutableUtmPoint vectorToUtm(Vector3d vector, MutableUtmPoint result) {
+        return geographicToUtm(vectorToGeographic(vector), result);
+    }
+
+    @Override
+    public MutableUtmPoint geographicToUtm(GeoPoint geoPoint, MutableUtmPoint result) {
+
+        ProjCoordinate sourceCoord = new ProjCoordinate(geoPoint.getLongitude(),geoPoint.getLatitude());
+        ProjCoordinate targetCoord = new ProjCoordinate();
+        CoordinateTransformFactory ctFactory = new CoordinateTransformFactory();
+
+        //UTMCRS created dynamically since it depends on the zone
+        zone = getUTMZone(geoPoint);
+        String utm_proj_str = "+proj=utm +zone=" + zone.number;
+        CRSFactory crsFactory = new CRSFactory();
+        CoordinateReferenceSystem utmCRS = crsFactory.createFromParameters("custom_proj", utm_proj_str);
+
+        CoordinateTransform transform = ctFactory.createTransform(wgs84CRS, utmCRS);
+
+        transform.transform(sourceCoord, targetCoord);
+        result.set(targetCoord.x, targetCoord.y, 0.0, getUTMZone(geoPoint));
```

Comment:
nit: `zone` from line 135 might be able to be reuse instead of called `getUTMZone(geoPoint)` again?

Reply:
Updated